ContainerProcessStrategy, allow running non-privileged agents in one container (#921)
DataDog/datadog-operator: docs/configuration.v2alpha1.md

| global.clusterAgentTokenSecret.keyName | KeyName is the key of the secret to use. |
| global.clusterAgentTokenSecret.secretName | SecretName is the name of the secret. |
| global.clusterName | ClusterName sets a unique cluster name for the deployment to easily scope monitoring data in the Datadog app. |

> | global.containerProcessStrategy.type | `type` sets a predetermined grouping of processes across containers. Two are supported at thist point: `singleProcessContainers`, default behavior, runs one process per container. `nonPrivilegedMultiProcessContainer`, runs non-privileged processes in a single container unless current configuration requires a privileged agent, for example `security-agent` or `system-probe` is required, fall back to singleProcessContainer Default: `singleProcessContainers` |

Reviewer: ```suggestion
| global.containerProcessStrategy.type | `type` sets a predetermined grouping of processes across containers. The two supported groupings: `singleProcessContainers`, default behavior, runs one process per container. `nonPrivilegedMultiProcessContainer`, runs non-privileged processes in a single container unless current configuration requires a privileged agent. For example `security-agent` or `system-probe` is required, falls back to `singleProcessContainer` because it's the default. |
```
